charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml

```diff
@@ -17,7 +17,9 @@ metadata:
 {{ toYaml .Values.prometheus.annotations | indent 4 }}
 {{- end }}
 spec:
+{{- if not (eq .Values.prometheus.prometheusSpec.automountServiceAccountToken nil) }}
```

Comment:
If I remember from last PR, this eq nil check require a minimum helm version?

Reply:
You're right! #4976 was the one about needing Helm 3.11.
I only remembered the subchart issue in #4818 😬
Compatibility issues everywhere!

Fixed in 7d8c6d1c.